chroma_core/lib/service_config.py

```diff
@@ -354,7 +355,8 @@ def _setup_influxdb(self):
                 settings.INFLUXDB_IML_STATS_DB,
                 "-execute",
                 "{}; {}; {}; {}".format(
```

Comment:
I'm getting an error on chroma-config setup after downgrading:

```
Creating InfluxDB database...
Traceback (most recent call last):
  File "/usr/bin/chroma-config", line 10, in <module>
    load_entry_point('iml-manager==6.0.0', 'console_scripts', 'chroma-config')()
  File "/usr/share/chroma-manager/chroma_core/lib/service_config.py", line 1056, in chroma_config
    errors = service_config.setup(username, password, ntpserver, check_db_space)
  File "/usr/share/chroma-manager/chroma_core/lib/service_config.py", line 794, in setup
    self._setup_influxdb()
  File "/usr/share/chroma-manager/chroma_core/lib/service_config.py", line 372, in _setup_influxdb
    settings.INFLUXDB_IML_STATS_DB, settings.INFLUXDB_IML_STATS_DB,
  File "/usr/share/chroma-manager/chroma_core/lib/util.py", line 192, in try_shell
    raise CommandError(cmdline, rc, out, err)
chroma_core.lib.util.CommandError: Command failed: ['influx', '-database', 'iml_stats', '-execute', 'CREATE CONTINUOUS QUERY "downsample_means" ON "iml_stats" BEGIN SELECT mean(*) INTO "iml_stats"."long_term".:MEASUREMENT FROM "iml_stats"."autogen"."target","iml_stats"."autogen"."host","iml_stats"."autogen"."node" GROUP BY time(30m),* END; CREATE CONTINUOUS QUERY "downsample_lnet" ON "iml_stats" BEGIN SELECT (last("send_count") - first("send_count")) / count("send_count") AS "mean_diff_send", (last("recv_count") - first("recv_count")) / count("recv_count") AS "mean_diff_recv" INTO "iml_stats"."long_term"."lnet" FROM "lnet" WHERE "nid" != \'"0@lo"\' GROUP BY time(30m),"host","nid" END; CREATE CONTINUOUS QUERY "downsample_samples" ON "iml_stats" BEGIN SELECT (last("samples") - first("samples")) / count("samples") AS "mean_diff_samples" INTO "iml_stats"."long_term"."target" FROM "target" GROUP BY time(30m),* END; CREATE CONTINUOUS QUERY "downsample_sums" ON "iml_stats" BEGIN SELECT (last("sum") - first("sum")) / count("sum") AS "mean_diff_sum" INTO "iml_stats"."long_term"."target" FROM "target" WHERE "units"=\'"bytes"\' GROUP BY time(30m),* END']
    return code 1
    stdout:
```

How should we handle the case where a CQ may have changed between releases?

Reply:
This is going to need some sort of migration framework to be correct.

There is no `ALTER CONTINUOUS QUERY` statement, which means CQs need to be dropped and recreated when there is a change. We don't want to do that for every update, otherwise we always wipe out historical data.